simpleflow/execute.py

```diff
@@ -181,7 +183,21 @@ def execute(*args, **kwargs):
                     close_fds=close_fds,
                     pass_fds=pass_fds,
                 )
-                rc = process.wait()
+
+                # Wait for process
+                if timeout:
```

Comment:
Can we isolate this code block elsewhere? `wait_subprocess(process, timeout)`? This is mostly aesthetics but this function is very long and I can't wrap my head around it tbh...